src/windows/cpu.cpp

```diff
@@ -120,6 +144,47 @@ int64_t CPU::currentClockSpeed_MHz() const {
   return speed[_core_id];
 }
 
+double CPU::currentUtility_Percentage() const {
+  std::vector<bstr_t> percentage{};
+  std::string& query = "Win32_PerfFormattedData_Counters_ProcessorInformation WHERE Name='" + std::to_string(_core_id) + ",_Total'";
+  wmi::queryWMI(query, "PercentProcessorUtility", percentage);
+  if (percentage.empty()) {
+    return -1.0;
+  }
+
+  const char* strValue = static_cast<const char*>(percentage[0]);
+  return std::stod(strValue);
+}
+
+double CPU::currentThreadUtility_Percentage(const int& thread_index) const {
+  std::vector<bstr_t> percentage{};
+  std::string& query = "Win32_PerfFormattedData_Counters_ProcessorInformation WHERE Name='" + std::to_string(_core_id) + "," + std::to_string(thread_index) + "'";
+  wmi::queryWMI(query, "PercentProcessorUtility", percentage);
+  if (percentage.empty()) {
+    return -1.0;
+  }
+
+  const char* strValue = static_cast<const char*>(percentage[0]);
```

Comment:
I am not sure, if the value can be null. If so, we would need an additional check...